src/mercury_engine_data_structures/formats/audio_common.py

```diff
@@ -0,0 +1,147 @@
+
+
+import construct
+from construct.core import (
+    Adapter,
+    Array,
+    Byte,
+    Const,
+    Construct,
+    Enum,
+    FocusedSeq,
+    Int16ul,
+    Int32ul,
+    Pointer,
+    Rebuild,
+    Struct,
+    Tell,
+)
+from construct.expr import Path
+from construct.lib.containers import Container, ListContainer
+
+from mercury_engine_data_structures.common_types import VersionAdapter, make_dict
+from mercury_engine_data_structures.construct_extensions.alignment import AlignTo
+
+# An int32 offset (or size) that can be rebuilt later in another struct using RebuildOffset().
+# stores the offset without needing extra Tells or private offsets cluttering the struct.
+RebuildableOffset = Struct(
```

Comment:
still have a strong suspicion that dunc will ask for a big refactor of this system whenever hii gets time to review it, lmao